checkfavors.lic

```diff
@@ -9,7 +9,8 @@ class CheckFavors
   def initialize
     @settings = get_settings
     town_data = get_data('town')
-    @hometown = town_data[@settings.hometown]
+    @character_hometown = @settings.favor_town || @settings.fang_cove_override_town || @settings.hometown
+    @hometown = town_data[@character_hometown]
     check_favors
```

Comment:
this would become
```ruby
check_favors(settings, hometown)
```
and the receiving method:
```ruby
def check_favors(settings, hometown)
```

Reply:
> this would become
> 
> ```ruby
> check_favors(settings, hometown)
> ```
> 
> and the receiving method:
> 
> ```ruby
> def check_favors(settings, hometown)
> ```

I don't fully understand. It sets @character_hometown to the first of favor_town, fang_cove_override_town, and hometown. It then sets @hometown to town_data[@character_hometown]. The only time a hometown is used in the execution is on line 27: 

`DRCT.walk_to(@hometown['favor_altar']['id'])`

Reply:
The point is that @hometown is accessible in other scripts, hometown is not. But that also means its not accessible outside that method, so you have to pass it. Making it a local variable eliminates interaction outside this script.

Reply:
It is @hometown in the original script. I wasn’t looking to alter anything about the original that wasn’t necessary to add in my variables.